UnityProject/Assets/Scripts/Systems/Score/ScoreMachine.cs

```diff
@@ -0,0 +1,183 @@
+
+		public enum ScoreType
+		{
+			Int,
+			Bool,
+			String,
+		}
+
+		public override void Awake()
+		{
+			base.Awake();
+			EventManager.AddHandler(Event.PreRoundStarted, () => Scores.Clear());
+			Scores = new Dictionary<string, ScoreEntry>(); //This only works if set on Awake()
+		}
+
+		/// <summary>
+		/// Used to add a new entry to the score machine to keep track of a subject. Make sure the ID is unique and saved somewhere!!!
+		/// </summary>
+		/// <param name="ID">The unique ID of the score entry. Used to grab and manipulate the entry from scores dictionary.</param>
+		/// <param name="scoreName">The name of the score shown on UIs</param>
+		/// <param name="type">Are you tracking a number? bool? or string?</param>
+		/// <param name="category">What category does this score fall under? (MiscScore does not appear on round end UI)</param>
+		/// <param name="alignment">Is entry considered a negative thing? a good thing? a weird thing? or do you not want to specify it's alignment?</param>
+		public static void AddNewScoreEntry(string ID, string scoreName, ScoreType type, ScoreCategory category = ScoreCategory.MiscScore, ScoreAlignment alignment = ScoreAlignment.Unspecified)
+		{
+			if (Instance.Scores.ContainsKey(ID))
+			{
+				Logger.LogWarning($"[ScoreMachine] - Attempting to add new entry with id ({ID}) but it already exists!");
+				return;
+			}
+			switch (type)
+			{
+				case ScoreType.Int:
+					ScoreEntryInt newEntryInt = new ScoreEntryInt
+					{
+						ScoreName = scoreName,
+						Category = category,
+						Alignment = alignment
+					};
+					Instance.Scores.Add(ID, newEntryInt);
+					break;
+				case ScoreType.Bool:
+					ScoreEntryBool newEntryBool = new ScoreEntryBool
+					{
+						ScoreName = scoreName,
+						Category = category,
+						Alignment = alignment
+					};
+					Instance.Scores.Add(ID, newEntryBool);
+					break;
+				case ScoreType.String:
+					ScoreEntryString newEntryString = new ScoreEntryString
+					{
+						ScoreName = scoreName,
+						Category = category,
+						Alignment = alignment
+					};
+					Instance.Scores.Add(ID, newEntryString);
+					break;
+				default:
+					throw new ArgumentOutOfRangeException(nameof(type), type, null);
+			}
+		}
+
+		/// <summary>
+		/// How much do you want to add or subtract from a score? (Use negative numbers to subtract from the score.)
+		/// </summary>
+		public static void AddToScoreInt(int valueToAddOnTop, string ID)
+		{
+			if (Instance.Scores.ContainsKey(ID) == false)
+			{
+				Logger.LogError($"[ScoreMachine] - {ID} does not exist in the score machine!");
+				return;
+			}
+
+			if (Instance.Scores[ID] is not ScoreEntryInt c)
+			{
+				Logger.LogError($"[ScoreMachine] - Attempted to add an integer to {ID} but it's entry is not a ScoreEntryInt!");
+				return;
+			}
+			c.Score += valueToAddOnTop;
+		}
+
+		/// <summary>
+		/// whats the new bool for this score?
+		/// </summary>
+		public static void AddToScoreBool(bool newValue, string ID)
```

Comment:
Can you use overloading here?

Reply:
Function overloading is the use when you have the same functions that does the same exact thing but with different arguments.
In the future, people might add custom types of scores that have different set of requirements and behaviors, so I've made a conscious decision to not use overloading and make the functions' functionality verbally written, so people can understand what use does these have for every thing.